osrf/vrx-docker
# Add 2023 mwe starter image example

## Purpose

This PR adds a minimal working example called `vrx_2023_starter`, analogous to the 2022 image. It's purpose is to give an example image in which the user builds a ROS 2 workspace.

As discussed in [VRX issue 719](https://github.com/osrf/vrx/issues/719), this will also make it easier to update the tutorial.

## To test
At least two things need to be tested. 
* First, check that the instructions work and the image builds and runs.
* Second, make sure images built using this process can be scored in the competition

### Build and Run
* Follow the instructions in the [README file](https://github.com/osrf/vrx-docker/blob/72effaa80cd5ab63240d7c56cfb64a23a369dfb2/mwes/2023/vrx_2023_starter/README.md) to build and run the example image.

### Check scoring
To make sure the resulting image is portable to other systems (despite Rocker customization), please also check that the image I built scored on your machine.
* Switch to the branch `M1chaelM/starter_test`.
* Verify that the `team_config` folder contains an additional example called `2023_starter_test`.
* Follow the [testing tutorial](https://github.com/osrf/vrx/wiki/vrx_2023-testing) using
  ```
  TEAM=2023_starter_test
  TASK=stationkeeping
  TRIAL=0
  ```
The test should run normally and the boat should move a little bit after the task enters the running state. As long as it works properly, this branch can be discarded.

## Bonus
I updated the `component_config.yaml` for the `example_team` to fix the parameter `sensor_name` for the pinger. This can be tested by running `./prepare_team_wamv.bash example_team` and checking that it produces a `urdf`.
